src/middlewares/authentication/base/authentication.middleware.ts

```diff
@@ -0,0 +1,34 @@
+import admin from 'firebase-admin';
+import { Request, Response, NextFunction } from 'express';
+
+import * as ERROR_MSG from '../../../constants/authErrorMessages';
+
+const AuthMiddleware = (permittedRoles: Array<String>) => (
```

Comment:
You mean the whole AuthMiddleware function or verifyIdToken?

Reply:
you can make authmiddleware async and use await inside

Reply:
If I use await like:

`await admin.auth().verifyIdToken(token) `

will I need to use .then and .catch after .verifyIdToken or not? I don't think so right? Since await should cover that already

Reply:
`try {
    const claims = await admin.stuff
    verify stuff
} catch e {
    handle error
}`

See aneesh's pr for an example